### PR TITLE
Fix for failing Molden writer tests.

### LINKED ISSUE #404: Molden writer tests not passing
I am getting failures in the new Molden tests when doing `python testall.py` with Open Babel installed via `pip` in my virtualenv.

```
======================================================================
ERROR: test_atoms_section_size (testmoldenwriter.MOLDENTest)                
Check if size of Atoms section is equal to expected.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "io/testmoldenwriter.py", line 43, in test_atoms_section_size
    writer = cclib.io.moldenwriter.MOLDEN(data)
  File "/media/Backup/virtualenvs/3/cclib_berquist/lib/python3.6/site-packages/cclib/io/filewriter.py", line 54, in __init__
    self.obmol, self.pbmol = self._make_openbabel_from_ccdata()
  File "/media/Backup/virtualenvs/3/cclib_berquist/lib/python3.6/site-packages/cclib/io/filewriter.py", line 73, in _make_openbabel_from_ccdata
    mult=self.ccdata.mult)
  File "/media/Backup/virtualenvs/3/cclib_berquist/lib/python3.6/site-packages/cclib/bridge/cclib2openbabel.py", line 45, in makeopenbabel
    obmol.SetTotalCharge(charge)
TypeError: in method 'OBMol_SetTotalCharge', argument 2 of type 'int'

======================================================================
ERROR: test_gto_section_size (testmoldenwriter.MOLDENTest)
Check if size of GTO section is equal to expected.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "io/testmoldenwriter.py", line 52, in test_gto_section_size
    writer = cclib.io.moldenwriter.MOLDEN(data)
  File "/media/Backup/virtualenvs/3/cclib_berquist/lib/python3.6/site-packages/cclib/io/filewriter.py", line 54, in __init__
    self.obmol, self.pbmol = self._make_openbabel_from_ccdata()
  File "/media/Backup/virtualenvs/3/cclib_berquist/lib/python3.6/site-packages/cclib/io/filewriter.py", line 73, in _make_openbabel_from_ccdata
    mult=self.ccdata.mult)
  File "/media/Backup/virtualenvs/3/cclib_berquist/lib/python3.6/site-packages/cclib/bridge/cclib2openbabel.py", line 45, in makeopenbabel
    obmol.SetTotalCharge(charge)
TypeError: in method 'OBMol_SetTotalCharge', argument 2 of type 'int'

======================================================================
ERROR: test_missing_attribute_error (testmoldenwriter.MOLDENTest)
Check if MissingAttributeError is raised as expected.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "io/testmoldenwriter.py", line 36, in test_missing_attribute_error
    cclib.io.moldenwriter.MOLDEN(data)
  File "/media/Backup/virtualenvs/3/cclib_berquist/lib/python3.6/site-packages/cclib/io/filewriter.py", line 54, in __init__
    self.obmol, self.pbmol = self._make_openbabel_from_ccdata()
  File "/media/Backup/virtualenvs/3/cclib_berquist/lib/python3.6/site-packages/cclib/io/filewriter.py", line 70, in _make_openbabel_from_ccdata
    obmol = makeopenbabel(self.ccdata.atomcoords,
AttributeError: 'ccData_optdone_bool' object has no attribute 'atomcoords'

======================================================================
ERROR: test_mo_section_size (testmoldenwriter.MOLDENTest)
Check if size of MO section is equal to expected.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "io/testmoldenwriter.py", line 68, in test_mo_section_size
    writer = cclib.io.moldenwriter.MOLDEN(data)
  File "/media/Backup/virtualenvs/3/cclib_berquist/lib/python3.6/site-packages/cclib/io/filewriter.py", line 54, in __init__
    self.obmol, self.pbmol = self._make_openbabel_from_ccdata()
  File "/media/Backup/virtualenvs/3/cclib_berquist/lib/python3.6/site-packages/cclib/io/filewriter.py", line 73, in _make_openbabel_from_ccdata
    mult=self.ccdata.mult)
  File "/media/Backup/virtualenvs/3/cclib_berquist/lib/python3.6/site-packages/cclib/bridge/cclib2openbabel.py", line 45, in makeopenbabel
    obmol.SetTotalCharge(charge)
TypeError: in method 'OBMol_SetTotalCharge', argument 2 of type 'int'

======================================================================
ERROR: test_molden_cclib_diff (testmoldenwriter.MOLDENTest)
Check if file written by cclib matched file written by Molden.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "io/testmoldenwriter.py", line 101, in test_molden_cclib_diff
    cclib_out = cclib.io.moldenwriter.MOLDEN(data).generate_repr()
  File "/media/Backup/virtualenvs/3/cclib_berquist/lib/python3.6/site-packages/cclib/io/filewriter.py", line 54, in __init__
    self.obmol, self.pbmol = self._make_openbabel_from_ccdata()
  File "/media/Backup/virtualenvs/3/cclib_berquist/lib/python3.6/site-packages/cclib/io/filewriter.py", line 73, in _make_openbabel_from_ccdata
    mult=self.ccdata.mult)
  File "/media/Backup/virtualenvs/3/cclib_berquist/lib/python3.6/site-packages/cclib/bridge/cclib2openbabel.py", line 45, in makeopenbabel
    obmol.SetTotalCharge(charge)
TypeError: in method 'OBMol_SetTotalCharge', argument 2 of type 'int'

----------------------------------------------------------------------
```

### PR DESCRIPTION
**Pull request by gaursagar**
This fixes #404

### CODE REVIEW

**Comment by berquist** [comment]
Looks good; Travis is failing because https://github.com/cclib/cclib/pull/395 hasn't been merged yet, not because of your PR.

**Comment by ATenderholt** [comment]
Do we ever expect non-integer charges in log files? If so, perhaps ```int(round(charge))``` would be better?

**Comment by gaursagar** [comment]
According to parsed data description in [docs](https://cclib.github.io/data.html#description-of-parsed-data) `ccdata.charge` would be an integer value.

**Comment by ATenderholt** [comment]
Then perhaps the GAMESS parser needs to be fixed too? See https://github.com/cclib/cclib/blob/master/src/cclib/parser/gamessparser.py#L1374.

**Comment by berquist** [comment]
@gaursagar, do you think you could track down why this is getting converted to a `float` from an `int` in the first place, so it can be fixed at the source?

**Comment by gaursagar** [comment]
@berquist Sure. I would take look.

**Comment by gaursagar** [comment]
@berquist as Adam had pointed out, `charge` was not being converted to `int` in the GAMESS parser. Other parsers don't have this issue. I have added a commit patching this issue.

File: src/cclib/parser/gamessparser.py [hunk 1]
```diff
@@ -1371,7 +1371,7 @@ def extract(self, inputfile, line):
             assert coords_and_charge.split()[-1] == '(A.U.)'
             reference = numpy.array([float(x) for x in coords_and_charge.split()[:3]])
             reference = utils.convertor(reference, 'bohr', 'Angstrom')
-            charge = float(coords_and_charge.split()[-2])
+            charge = int(float(coords_and_charge.split()[-2]))
```
Comment (ATenderholt) [inline src/cclib/parser/gamessparser.py]: I think there should be a ```round``` here. If the string is something like ```1.9999```, then the charge will incorrectly be determined as ```1```:

**Comment by berquist** [comment]
I thought that it would be converted to the correct type during our "type checking" phase (see [here](https://github.com/cclib/cclib/blob/80ee63b6451623f7566d9405a65d924f31d68cce/src/cclib/parser/data.py#L259)). For example, some parsers set attributes as lists of lists or lists of arrays, which is fine because the final attribute check/removal converts them to the correct type.

My point is that GAMESS parsing a float shouldn't matter, because it should be automatically converted later.

**Comment by gaursagar** [comment]
@berquist I believe a `setattr` is missing in the `try` block.
```python
try:
    val = self._attributes[attr].type(val)
    setattr(self, attr, val)
```

**Comment by gaursagar** [comment]
The last commit I added https://github.com/cclib/cclib/pull/405/commits/6dbac041d1cfc89de2737aef41691ffd73566d80 fixes converted attributes not being set after typecheck.

(Some other lines got changed on their own due to mixed line-endings in `data.py`)

**Comment by gaursagar** [comment]
I have reverted changes to `data.py`, Karol would be taking care of it (`typecheck` was interferring with the `--future` flag behaviour of `optdone`).

Other changes ensure that charge is parsed as `int` in GAMESS and that Molden writer does not fail tests with OpenBabel installed.

**Comment by langner** [comment]
> I have removed commit which was changing data.py. Karol would be taking care of it (typecheck was interferring with the --future flag behaviour of optdone).

Please see #408 for that

**Pull request merged**